lib/parser/parse-tree.cc

```diff
@@ -86,14 +86,17 @@ bool DoConstruct::IsDoConcurrent() const {
   return control && std::holds_alternative<LoopControl::Concurrent>(control->u);
 }
 
-static Designator MakeArrayElementRef(
-    const Name &name, std::list<Expr> &&subscripts) {
+static Designator MakeArrayElementRef(const Name &name,
+    const parser::CharBlock &source, std::list<Expr> &&subscripts) {
   ArrayElement arrayElement{DataRef{Name{name}}, std::list<SectionSubscript>{}};
   for (Expr &expr : subscripts) {
     arrayElement.subscripts.push_back(SectionSubscript{
         Scalar{Integer{common::Indirection{std::move(expr)}}}});
   }
-  return Designator{DataRef{common::Indirection{std::move(arrayElement)}}};
+  auto result{
```

Comment:
This replacement of `source` could be done in the caller or in a wrapper; it seems odd to do it here.